### PR TITLE
Ping.

### PR DESCRIPTION
**Pull request by boguslaw-wojcik**

### CODE REVIEW

File: adapter/network_management.md [hunk 1]
```diff
@@ -251,3 +274,55 @@ Every adapter should define its own service name.
...
+```
+
+* Example of a report containing ping report for thing 4 connected indirectly through single hop at address 2:
+
+```json
+{
+  "serv": "zwave-ad",
+  "type": "evt.ping.report",
+  "val_t": "object",
+  "val": {
```
Comment (eternalamenity) [inline adapter/network_management.md:303]: What about being backward compatible and having more proeprties here for FimpUI usage as an example?
We want to keep it backward compatible, but we do not want to state it in FIMP specs?

File: adapter/network_management.md [hunk 2]
```diff
@@ -73,6 +75,23 @@ Every adapter should define its own service name.
...
+| delay   | int    | `321`                                            | Number of milliseconds between sending a ping and receiving a response or acknowledgement.                                                                    |
+| nodes   | object | `[{"address":"4","type":"rssi","value":"-87"}]"` | An optional array of node connection quality, where the first element represents the hub while the last one the thing. See [`routing_quality`](#definitions). |
+
+* `routing_quality` is an object with the following structure:
+
+| Field   | Type   | Example  | Description                                                     |
+|---------|--------|----------|-----------------------------------------------------------------|
+| address | string | `"4"`    | Unique address of a thing.                                      |
+| type    | string | `"rssi"` | Type of connection quality measurement, either `rssi` or `lqi`. |
+| value   | string | `"-87"`  | Value of connection quality.                                    |
```
[the remark this answers is not in the document]
Comment (boguslaw-wojcik) [reply adapter/network_management.md:93]: Optional.

File: adapter/network_management.md [hunk 3]
```diff
@@ -43,8 +43,10 @@ Every adapter should define its own service name.
 | in   | cmd.network.node_update      | string¹    | Requests an interview and discovery of its capabilities for the provided address of a thing.                      |
 | in   | cmd.network.reset            | null       | Requests forceful removal of all things and resets the network.                                                   |
 | out  | evt.network.reset_done       | null       | Reports that network reset has been completed.                                                                    |
+| in   | cmd.ping.send                | string     | Requests execution of a ping for the provided address of a thing.                                                 |
```
Comment (BudzikM) [inline adapter/network_management.md:46]: adress is a string ? Havent you changed it recently from string to int somwhere?

File: adapter/network_management.md (hunk 1 above)
Comment (boguslaw-wojcik) [reply adapter/network_management.md:303]: Yes. Leave just comments. Also Angry dog uses only success/failure and only TP flow is using all the bs left in Z-Wave message.

File: adapter/network_management.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (boguslaw-wojcik) [reply adapter/network_management.md:93]: You can provide just one LQI here or not at all, or in later iterations. The most important thing is that you look if the structure works for you.

File: adapter/network_management.md [hunk 4]
```diff
@@ -73,6 +75,23 @@ Every adapter should define its own service name.
 
 * `network_update_mode` is one of the following values: `full`, `topology`.
 
+* `ping_report` is an object with the following structure:
+
+| Field   | Type   | Example                                          | Description                                                                                                                                                   |
+|---------|--------|--------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| address | string | `"4"`                                            | Unique address of a thing.                                                                                                                                    |
+| status  | string | `"SUCCESS"`                                      | Either `SUCCESS` for a successful ping or `FAILED` if response was not received within configured timeout.                                                    |
```
[the remark this answers is not in the document]
Comment (BudzikM) [reply adapter/network_management.md:83]: If we need rsp maybe it'd be better to use error rsp msg ?

[the remark this answers is not in the document]
Comment (boguslaw-wojcik) [reply adapter/network_management.md:83]: Nope.
* We need to maintain BC with what is in Z-Wave and used by some clients.

File: adapter/network_management.md (hunk 3 above)
Comment (boguslaw-wojcik) [reply adapter/network_management.md:46]: Nope, from int to string.You got confused.

**Review by eternalamenity — APPROVED** [review]

**Review by budzonmichal — APPROVED** [review]

**Review by BudzikM — APPROVED** [review]

**Pull request merged**